community/kernel-api/src/test/java/org/neo4j/internal/kernel/api/RelationshipTransactionStateTestBase.java

```diff
@@ -299,4 +280,113 @@ private void traverseWithoutGroups( RelationshipTestSupport.StartNode start, boo
+                    }
+                    else
+                    {
+                        group.outgoing( relationship );
+                    }
+                    // then
+                    assertCount( session, relationship, expectedCounts, group.relationshipLabel(), OUTGOING );
+
+                    // incoming
+                    if ( detached )
+                    {
+                        read.relationships( start.id, group.incomingReference(), relationship );
+                    }
+                    else
+                    {
+                        group.incoming( relationship );
+                    }
+                    // then
+                    assertCount( session, relationship, expectedCounts, group.relationshipLabel(), INCOMING );
+
+                    // loops
+                    if ( detached )
+                    {
+                        read.relationships( start.id, group.loopsReference(), relationship );
+                    }
+                    else
+                    {
+                        group.loops( relationship );
+                    }
+                    // then
+                    assertCount( session, relationship, expectedCounts, group.relationshipLabel(), BOTH );
+                }
+            }
+        }
+    }
+
+    private Map<String,Integer> modifyStartNodeRelationships( RelationshipTestSupport.StartNode start, Transaction tx )
+            throws KernelException
+    {
+        Map<String, Integer> expectedCounts = new HashMap<>();
+        for ( Map.Entry<String,List<RelationshipTestSupport.StartRelationship>> kv : start.relationships.entrySet() )
+        {
+            List<RelationshipTestSupport.StartRelationship> rs = kv.getValue();
+            RelationshipTestSupport.StartRelationship head = rs.get( 0 );
+            int label = session.token().relationshipType( head.type.name() );
+            switch ( head.direction )
+            {
+            case INCOMING:
+                tx.dataWrite().relationshipCreate( tx.dataWrite().nodeCreate(), label, start.id );
+                tx.dataWrite().relationshipCreate( tx.dataWrite().nodeCreate(), label, start.id );
+                break;
+            case OUTGOING:
+                tx.dataWrite().relationshipCreate( start.id, label, tx.dataWrite().nodeCreate() );
+                tx.dataWrite().relationshipCreate( start.id, label, tx.dataWrite().nodeCreate() );
+                break;
+            case BOTH:
+                tx.dataWrite().relationshipCreate( start.id, label, start.id );
+                tx.dataWrite().relationshipCreate( start.id, label, start.id );
+                break;
+            default:
+                throw new IllegalStateException( "Oh ye be cursed, foul checkstyle!" );
+            }
+            tx.dataWrite().relationshipDelete( head.id );
+            expectedCounts.put( kv.getKey(), rs.size() + 1 );
+        }
+
+        String newLabelName = "NEW";
```

Comment:
I know I'll probably push your buttons here, but can we try to limit `label` for relationship type to a minimum? It looks especially weird since the methods has the term `relationshipType`. A variable called something... label primes the mind that it is a node label and you'll have to look at usages to figure out whether it's a node label or relationship type.